main/manager.c

```diff
@@ -6873,6 +6900,203 @@ static int action_coreshowchannels(struct mansession *s, const struct message *m
+        }
+
+	size = strlen(s) + 1;
+	str = ast_malloc(size);
+	if (!str) {
+		ast_log(LOG_ERROR, "Unable to append channel to channel map\n");
+		return 1;
+	}
+
+	ast_copy_string(str, s, size);
+	if (AST_VECTOR_APPEND(v, str)) {
+		ast_log(LOG_ERROR, "Unable to append channel to channel map\n");
+		ast_free(str);
+		return 1;
+	}
+
+	return 0;
+}
+
+/*! \brief Recursive function to get all channels in a bridge. Follow local channels as well */
+static int coreshowchannelmap_add_connected_channels(struct ast_vector_string *channel_map,
+	struct ast_channel_snapshot *channel_snapshot, struct ast_bridge_snapshot *bridge_snapshot)
+{
+	int res = 0;
+	struct ao2_iterator iter;
+	char *current_channel_uid;
+
+	iter = ao2_iterator_init(bridge_snapshot->channels, 0);
+	while ((current_channel_uid = ao2_iterator_next(&iter))) {
+		struct ast_channel_snapshot *current_channel_snapshot;
+		int add_channel_res;
+
+		/* Don't add the original channel to the list - it's either already in there,
+		 * or it's the channel we want the map for */
+		if (!strcmp(current_channel_uid, channel_snapshot->base->uniqueid)) {
+			continue;
+		}
+
+		current_channel_snapshot = ast_channel_snapshot_get_latest(current_channel_uid);
+		if (!current_channel_snapshot) {
+			ast_log(LOG_ERROR, "Unable to get snapshot of channel\n");
+			res = 1;
+			break;
+		}
+
+		add_channel_res = coreshowchannelmap_add_to_map(channel_map, current_channel_snapshot->base->name);
+		if (add_channel_res > 0) {
+			ast_log(LOG_ERROR, "Unable to add channel to vector\n");
+			res = 1;
+			ao2_ref(current_channel_snapshot, -1);
+			break;
+		}
+
+		/* If this is a local channel that we haven't seen yet, let's go ahead and find out what else is connected to it */
+		if (!add_channel_res && ast_begins_with(current_channel_snapshot->base->name, "Local")) {
+			struct ast_channel_snapshot *other_local_snapshot;
+			struct ast_bridge_snapshot *other_bridge_snapshot;
+			char *other_local;
+			int size = strlen(current_channel_snapshot->base->name);
+
+			other_local = ast_malloc(size + 1);
+			if (!other_local) {
+				ast_log(LOG_ERROR, "Unable to allocate space for other local channel string\n");
+				ao2_ref(current_channel_snapshot, -1);
+				return 1;
+			}
+
+			/* Don't copy the trailing number - set it to 1 or 2, whichever one it currently is not */
+			ast_copy_string(other_local, current_channel_snapshot->base->name, size);
+			other_local[size - 1] = ast_ends_with(current_channel_snapshot->base->name, "1") ? '2' : '1';
+			other_local[size] = '\0';
+
+			other_local_snapshot = ast_channel_snapshot_get_latest_by_name(other_local);
+			if (!other_local_snapshot) {
+				ast_log(LOG_ERROR, "Unable to get other local channel\n");
+				ao2_ref(current_channel_snapshot, -1);
+				ast_free(other_local);
+				return 1;
+			}
+
+			if (coreshowchannelmap_add_to_map(channel_map, other_local_snapshot->base->name)) {
+				ast_log(LOG_ERROR, "Unable to add channel to vector\n");
```

Comment:
Is this log also redundant? If res > 0, something is already logged, and if res == -1, it was already in the vector so this is misleading.